lib/web_ui/lib/src/engine/html/canvas.dart

```diff
@@ -288,6 +288,47 @@ class SurfaceCanvas implements ui.Canvas {
     _canvas.drawImageRect(image, src, dst, paint as SurfacePaint);
   }
 
+  // Return a list of slice coordinates based on the size of the nine-slice parameters in
+  // one dimension. Each set of slice coordinates contains a begin/end pair for each of the
+  // source (image) and dest (screen) in the order (src0, dst0, src1, dst1).
+  // The area from src0 => src1 of the image is painted on the screen from dst0 => dst1
+  // The slices for each dimension are generated independently.
+  List<double> _initSlices(double img0, double imgC0, double imgC1, double img1, double dst0, double dst1) {
+    final double imageDim = img1 - img0;
+    final double destDim = dst1 - dst0;
+
+    if (imageDim == destDim) {
+      // If the src and dest are the same size then we do not need scaling
+      // We return 4 values for a single slice
+      return [ img0, dst0, img1, dst1 ];
```

Comment:
Let's specify generic arguments. We have a bug in our analysis options to enforce this, but we try to conform to the Flutter style guide for Dart.

Reply:
I have that staged to go, but I'm waiting to update the goldens commit hash.